kube-core/src/params.rs

```diff
@@ -103,18 +103,22 @@ impl ListParams {
         }
         if let Some(continue_token) = &self.continue_token {
             qp.append_pair("continue", continue_token);
-        }
-
-        if let Some(rv) = &self.resource_version {
-            qp.append_pair("resourceVersion", rv.as_str());
-        }
-        match &self.version_match {
-            None => {}
-            Some(VersionMatch::NotOlderThan) => {
-                qp.append_pair("resourceVersionMatch", "NotOlderThan");
-            }
-            Some(VersionMatch::Exact) => {
-                qp.append_pair("resourceVersionMatch", "Exact");
+        } else {
+            // When there's a continue token, we don't want to set resourceVersion
+            if let Some(rv) = &self.resource_version {
+                if rv != "0" || (rv == "0" && self.limit.is_none()) {
+                    qp.append_pair("resourceVersion", rv.as_str());
```

Comment:
Going over this again just as a sanity; is this possibly slightly imprecise?

If `resourceVersion` is `"0"` with no limit, we end up NOT serialising `resourceVersion` which puts us in "Most Recent" semantics rather than "Any" semantics. Presumably this was not intended and the `resourceVersion` should be serialized above the first if?

Reply:
Maybe we need to add a test for that too, but as per the snippet above, if rv=0 and the limit is unset, the resourceVersion is serialized.